src/sql/src/plan/statement/ddl.rs

```diff
@@ -658,8 +696,23 @@ pub fn plan_create_source(
                 // column and casts it to the appropriate target type
                 let mut column_casts = vec![];
                 for (i, column) in table.columns.iter().enumerate() {
-                    let ty = mz_pgrepr::Type::from_oid_and_typmod(column.type_oid, column.type_mod)
-                        .map_err(|e| sql_err!("{}", e))?;
+                    let ty = match text_cols_dict.get(&Oid(table.oid)) {
+                        Some(names) if names.contains_key(&column.name) => mz_pgrepr::Type::Text,
+                        _ => {
+                            match mz_pgrepr::Type::from_oid_and_typmod(
+                                column.type_oid,
+                                column.type_mod,
+                            ) {
+                                Ok(t) => t,
+                                // Skipping this table cast is OK because we
+                                // have validated that this table from the
+                                // publication is not being used during
+                                // purification
+                                Err(_) => continue 'table,
```

Comment:
Hm, the resulting `column_casts` seems a bit brittle as it doesn't have the same shape for all the tables in the publication. I would suggest always populating the cast expression and when we can't find the type we can just put an expression that calls `mz_error_if_null(NULL, "Unsupported type with OID {column.type_oid}")`

Reply:
Sure thing.